File: 0386-lexicographical-numbers/0386-lexicographical-numbers.cpp
class Solution {
private:
    int n;
    void backtrack(int curr, vector<int>& ans){
        if (curr > n) return;

        ans.push_back(curr);

        for (int i = 0; i < 10; ++i)
            backtrack(curr * 10 + i, ans);
    }

public:
    vector<int> lexicalOrder(int n) {
        this -> n = n;
        vector<int> ans;
        for (int i = 1; i < 10; ++i)
            backtrack(i, ans);

        return ans;
    }
};

File: 0386-lexicographical-numbers/0386-lexicographical-numbers.py
class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        def backtrack(curr: int) -> None:
            if curr > n:
                return

            ans.append(curr)
            for i in range(0, 10):
                backtrack(curr * 10 + i)

        ans = []
        for i in range(1,10):
            backtrack(i)

        return ans